satpy/modifiers/angles.py

```diff
@@ -358,14 +358,16 @@ def get_satellite_zenith_angle(data_arr: xr.DataArray) -> xr.DataArray:
     return satz
 
 
-def get_cos_sza(data_arr: xr.DataArray) -> xr.DataArray:
+def get_cos_sza(data_arr: xr.DataArray, chunks: Union[int, str, tuple] = None) -> xr.DataArray:
     """Generate the cosine of the solar zenith angle for the provided data.
 
     Returns:
         DataArray with the same shape as ``data_arr``.
 
     """
-    lons, lats = _get_valid_lonlats(data_arr.attrs["area"], data_arr.chunks)
+    if chunks is None:
+        chunks = data_arr.chunks
+    lons, lats = _get_valid_lonlats(data_arr.attrs["area"], chunks)
```

Comment:
The point being that you could get rid of the `chunks` keyword argument and the `chunks =` in the DayNightCompositor and just pass the DataArray itself.

Reply:
Yes. It would mean the signature of `get_cos_sza` (the arguments that it takes) would be unchanged. It also simplifies your calling of it in the D/N compositor.

Reply:
Ok, I've done this.